privacyidea/lib/eventhandler/requestmangler.py

```diff
@@ -0,0 +1,163 @@
+                       {"parameter":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The parameter that should be deleted.")}
+                        },
+                   ACTION_TYPE.ADD:
+                       {"parameter":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The parameter that should be added or modified.")
+                             },
+                        "value":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The new value of the parameter. Can contain tags like {0}, {1} for  "
+                                              "the matched sub strings.")
+                            },
+                        "match_parameter":
+                            {"type": "str",
+                             "description": _("The parameter, that should match some values.")
+                            },
+                        "match_pattern":
+                            {"type": "str",
+                             "description": _("The value of the match_parameter. It can contain a regular "
+                                              "expression and '()' to transfer values to the new parameter.")
+                            }
+                        }
+                   }
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action:
+        :param options: Contains the flask parameters g, request, response
+            and the handler_def configuration
+        :type options: dict
+        :return:
+        """
+        ret = True
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options", {})
+        parameter = handler_options.get("parameter")
+
+        if parameter:
+            if action.lower() == ACTION_TYPE.DELETE:
+                if parameter in request.all_data:
+                    del(request.all_data[parameter])
+            elif action.lower() == ACTION_TYPE.ADD:
+                value = handler_options.get("value")
+                match_parameter = handler_options.get("match_parameter")
+                match_pattern = handler_options.get("match_pattern")
+                if value:
+                    # We only take action, if we have a value
+                    if not match_parameter:
+                        # simple setting a parameter
+                        request.all_data[parameter] = value
+                    elif match_pattern:
+                        # setting a parameter depending on another value
+                        m = re.match(match_pattern, request.all_data.get(match_parameter))
```

Comment:
This will cause an exception ``TypeError: expected string or buffer``  if ``request.all_data`` does not contain ``match_parameter``. What should we do in that case?

Maybe we should add a flag ``abort_on_error`` that describes what to do in case of error?

Reply:
So you say, I need to check if ``request.all_data(match_parameter)`` is ``None``.

I think in this case, we simply will not modify the other parameter.

Reply:
done in 6cc4d4a2